### PR TITLE
Chunks/enum interface

### PR DESCRIPTION
**Pull request by cv3d**
Please merge with opencv/opencv#12310

### This pullrequest changes
Refer to opencv/opencv#12288 for details

```
force_builders=Custom,linux32,win32,oclmacosx,windows_vc15
docker_image:Custom=ubuntu-cuda:16.04
allow_multiple_commits=1
```

### CODE REVIEW

File: modules/xfeatures2d/include/opencv2/xfeatures2d.hpp [hunk 1]
```diff
@@ -84,10 +84,13 @@ class CV_EXPORTS_W FREAK : public Feature2D
...
 public:
 
-    enum
-    {
-        NB_SCALES = 64, NB_PAIRS = 512, NB_ORIENPAIRS = 45
-    };
+    static const double LOG2;
+    static const int    NB_ORIENTATION   = 256;
+    static const int    NB_POINTS        = 43;
+    static const int    SMALLEST_KP_SIZE = 7; // smallest size of keypoints
```
[the remark this answers is not in the document]
Comment (cv3d) [reply modules/xfeatures2d/include/opencv2/xfeatures2d.hpp]: @alalek Ok, done!

**Comment by cv3d** [comment]
@alalek I am sorry to ask for revision again as I CUDA enums got moved here..

**Comment by alalek** [comment]
Please take a look on failed builds.

**Comment by cv3d** [comment]
@alalek All green now!

**Pull request merged**